fedimint/src/config.rs

```diff
@@ -45,21 +49,32 @@ pub struct ServerConfig {
 
 #[derive(Debug, Clone, Serialize, Deserialize)]
 pub struct Peer {
-    pub connection: ConnectionConfig,
+    pub hbbft: ConnectionConfig,
     #[serde(with = "serde_tls_cert")]
     pub tls_cert: rustls::Certificate,
+    /// The peer's websocket network address and port (e.g. `ws://10.42.0.10:5000`)
+    pub api_addr: Url,
 }
 
-#[derive(Debug)]
+#[derive(Debug, Clone)]
+/// network config for a server
 pub struct ServerConfigParams {
-    pub hbbft_base_port: u16,
-    pub api_base_port: u16,
+    pub tls: TlsConfig,
+    pub hbbft: NetworkConfig,
+    pub api: NetworkConfig,
+    pub server_dkg: NetworkConfig,
+    pub wallet_dkg: NetworkConfig,
+    pub lightning_dkg: NetworkConfig,
+    pub mint_dkg: NetworkConfig,
```

Comment:
Can't they all use the same connection manager? Just have an enum for the module-specific message types?

Reply:
You mean reuse the same enum for all modules until modularization happens?

Reply:
I think that's what I mean? Like, we could have an

```rust
enum ConfigGenMessage {
    Federation(…),
    Wallet(…),
    Mint(…),
    LN(…),
}

that will later be replaced with a `ConfigGenMessage(Box<dyn IConfigGenMessage>)` just like the other enums of this kind.

Reply:
We should probably chat about the best way to do that because it will need to change how `PeerConnections` works (or maybe we create wrappers for it?).